hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/auth/delegation/AbstractDelegationTokenBinding.java

```diff
@@ -157,17 +159,19 @@ public Text getOwnerText() {
    * This will only be called if a new DT is needed, that is: the
    * filesystem has been deployed unbonded.
    *
-   * If {@link #createDelegationToken(Optional, EncryptionSecrets)}
+   * If {@link #createDelegationToken(Optional, EncryptionSecrets, Text)}
    * is overridden, this method can be replaced with a stub.
    *
    * @param policy minimum policy to use, if known.
    * @param encryptionSecrets encryption secrets for the token.
+   * @param renewer the principal permitted to renew the token.
    * @return the token data to include in the token identifier.
    * @throws IOException failure creating the token data.
    */
   public abstract AbstractS3ATokenIdentifier createTokenIdentifier(
       Optional<RoleModel.Policy> policy,
-      EncryptionSecrets encryptionSecrets) throws IOException;
+      EncryptionSecrets encryptionSecrets,
+      Text renewer) throws IOException;
```

Comment:
As these are public methods, are you sure that we don't need to preserve the old signature as well for possible extensions? Does this appear in a public release yet at all?

Reply:
This has not yet appeared in a public release; Steve actually suggested that I remove the previously-existing signatures, which I had marked as deprecated initially.

Reply:
no ASF release. The only people using this outside the hadoop-aws jar are you; I have plans to do other incompatible changes soon. Sorry.